micro-rdk-installer/src/nvs/request.rs

```diff
@@ -0,0 +1,146 @@
+use super::data::ViamFlashStorageData;
+
+use rcgen::{date_time_ymd, CertificateParams, DistinguishedName};
+
+use local_ip_address::local_ip;
+use tokio::runtime::Runtime;
+use viam::gen::proto::app::v1::{
+    robot_service_client::RobotServiceClient, AgentInfo, CertificateRequest, ConfigRequest,
+};
+use viam_rust_utils::rpc::dial::{DialOptions, RPCCredentials};
+
+/*
+This module contains the logic for acquiring the security credentials for a robot from
+the Viam App and preparing for flash storage
+*/
+
+pub fn populate_nvs_storage_from_app(
+    storage_data: &mut ViamFlashStorageData,
+) -> anyhow::Result<()> {
+    populate_dtls_certificate(storage_data)?;
+    let rt = Runtime::new()?;
+    rt.block_on(read_cloud_config(storage_data))?;
+    rt.block_on(read_certificates(storage_data))?;
```

Comment:
nit: these functions are called `read_*`, but don't actually return anything. maybe rename to `set_*` or `process_*`?